### PR TITLE
PR for Search Video Functionality

### PR DESCRIPTION
**Pull request by SK02K1**
# Search video functionality added

# Features
- Search and filter videos based on the search query.
- Show no video found if no video found based on search query.

Screenshot 2022-04-02 at 9 31 11 PM
Screenshot 2022-04-02 at 9 31 40 PM

### CODE REVIEW

File: src/pages/Search/Search.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+import { SearchBar, Loader, VideoDetailsCard } from '../../components';
+import { useVideosData } from '../../contexts';
+import { useAxios } from '../../hooks';
+import { VIDEOS_ACTIONS } from '../../utils';
+import { filterVideosBySearchQuery } from '../../utils';
+
+export const Search = () => {
+  const [searchQuery, setSearchQuery] = useState('');
```
Comment (xsaurabhsuthar) [inline src/pages/Search/Search.jsx:9]: Suggestion: You can move the component to logic to a separate Container like SearchContainer and use it in the View component.

File: src/pages/Search/Search.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      <SearchBar query={searchQuery} handleQueryChange={updateSearchQuery} />
+      <div className='grid-container auto m-md-tb'>
+        {filteredVideos.map((videoDetails) => (
+          <VideoDetailsCard
+            key={videoDetails._id}
+            videoDetails={videoDetails}
+          />
+        ))}
+      </div>
+      {searchQuery && !Boolean(filteredVideos.length) && (
```
Comment (xsaurabhsuthar) [inline src/pages/Search/Search.jsx:43]: Good use of Boolean for type checking

**Review by xsaurabhsuthar — APPROVED** [review]
Everything is very clean and well written. Good to merge🚀🚀🚀🚀

**Review by manismk — APPROVED** [review]
Looks clean and Good.
Good naming convention.

**Pull request merged**